packages/blockchain-wallet-v4-frontend/src/data/modules/profile/sagas.ts

```diff
@@ -589,6 +541,58 @@ export default ({ api, coreSagas, networks }) => {
     }
   }
 
+  const signIn = function* ({ payload }) {
+    try {
+      const { firstLogin } = payload
+      const email = (yield select(selectors.core.settings.getEmail)).getOrFail('No email')
+      const guid = yield select(selectors.core.wallet.getGuid)
+      // TODO: in future only fetch unified credentials
+      const unifiedNabuCredentialsR = yield select(
+        selectors.core.kvStore.unifiedCredentials.getNabuCredentials
+      )
+      const userCredentialsR = yield select(
+        selectors.core.kvStore.userCredentials.getLegacyNabuCredentials
+      )
+      if (!Remote.Success.is(unifiedNabuCredentialsR)) {
+        yield call(coreSagas.kvStore.unifiedCredentials.fetchMetadataUnifiedCredentials)
+      }
+      if (!Remote.Success.is(userCredentialsR)) {
+        yield call(coreSagas.kvStore.userCredentials.fetchMetadataUserCredentials)
+      }
+      const { nabuLifetimeToken, nabuUserId } = (yield select(
+        selectors.core.kvStore.unifiedCredentials.getUnifiedOrLegacyNabuEntry
+      )).getOrElse({})
+      // sync legacy nabu credentials with unified kv entry
+      const unifiedNabuCredentials = (yield select(
+        selectors.core.kvStore.unifiedCredentials.getNabuCredentials
+      )).getOrElse({})
+      if (!unifiedNabuCredentials.nabuUserId || !unifiedNabuCredentials.nabuLifetimeToken) {
+        yield put(
+          actions.core.kvStore.unifiedCredentials.setUnifiedCredentials({
+            nabu_lifetime_token: nabuLifetimeToken,
+            nabu_user_id: nabuUserId
+          })
+        )
+      }
+      if (!firstLogin && (!nabuUserId || !nabuLifetimeToken)) {
```

Comment:
Main change is here, call `/users` if there's no nabuID in metadata. The error in the screenshot was being thrown at 581.